design/design.md

```diff
@@ -0,0 +1,414 @@
+To make sure that the chosen block's payload is correct and available
+(which potentially could be not due to malicious behavior),
+all blocks in the input to the consensus protocol need to have
+a certificate of availability proving that at least one honest node has verified
+the block content and is available to distribute it.
+A natural solution to certify data availability would be to use a
+weighted threshold multisignature scheme constructed using decentralized ALBA.
+
+Cross-chain bridges are protocols allowing one blockchain to verify
+information on another blockchain.
+In essense this is the same problem that the Mithril solves with a potentially
+added constraint that on-chain verification must be computationally cheap.
+Whereas Mithril is specific to the Cardano blockchain, one can envision
+a similar protocol, powered by ALBA, for other cryptocurrencies including
+partner chains like Midnight.
+
+Applications of ALBA are not limited to Cardano and blockchains in general.
+For instance, the [paper](https://eprint.iacr.org/2023/1655) describes how
+this primitive can be used to optimize the prover running time of
+a Universally Composable SNARK.
+
+## Requirements
+
+We intend to make an implementation of ALBA that can be used in
+Mithril, Peras, Leios, cross-chain bridges and potentially other protocols
+inside and outside the Cardano universe.
+Our implementation will mostly be guided by our main "clients":
+Mirthril, Peras and Leios.
+
+ALBA is a generic tool that lets one prove knowledge of any kind of data
+elements, but
+as noted in (todo: alba applications section), these three applications
+don't need the full expressivity of ALBA, they really need
+a multisignature scheme, where the data elements would specifically be
+signatures.
+Thus there is question of whether we want our library to provide a general
+ALBA interface or simply an interface for a multisignature scheme.
+We will do both.
+1. We will implement the most generic version of ALBA to make it suitable for
+as many applications as possible.
+This means that there is only a generic notion of "data elements" and no notion
+of signatures, VRF, etc.
+2. We will use the generic ALBA library to implement a multisignature scheme,
+either as a small example in this repository or a full-fledged library in a
+separate repository.
+Either way, the multi-signature implementation would serve as a flagship
+example of how the ALBA library can be used.
+
+The requirements for our (general) ALBA implementation can be roughly split
+in four groups: 
+1. definitions: what do the ALBA schemes we implement guarantee, under what
+assumptions?
+2. correctness: does our implementation provide the same guarantees of
+soundness and completeness as the ALBA schemes on paper?
+3. performance: proof size, communication complexity, computational complexity.
+4. interfaces: the applications should be able to use our
+implementation relatively easily
+
+### Definitions
```

Comment:
I do not understand what you mean by "definitions". API? Interface? Be more precise and clear

Reply:
I am not fond of the title (as it is not clear) and not sure about what you say. Is it that you want the design of the implementation to be generic enough?

Reply:
So you are saying that the repository should be an accurate implementation of the paper? You could rename this paragraph then "Consistency (to the paper/specifications)"